docs/how_to/build-prim-based.rst

```diff
@@ -0,0 +1,109 @@
+Primitive-based routines
+========================
+
+We consider primitive-based routines that use the primitives interfaces
+but perform more complicated actions, such as fidelity and gradient
+routines, and povm measurement. They accept basic primitives (Sampler
+or Estimator) in the initializer and use them for calculations.
+
+Fidelity routine
+------------------
+
+Suppose that we have an algorithm that requires a calculation of quantum
+states overlaps. To save us some work, we might use the higher-level
+fidelity routine, such as pVQD or VQD, that accepts the Sampler
+primitive.
+
+In the code example, we can see how a user
+would implement this algorithm with the quantum instance in a case that
+covers both the statevector simulator and a shot-based backend.
+
+
+Code example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   from qiskit.algorithms.state_fidelities import BaseStateFidelity
+   from qiskit import QuantumCircuit
+
+   #fidelity = ComputeUncompute(Sampler())
+   class QuantumFidelityAlgorithm:
```

Comment:
I didn't understand the purpose of this how-to until after reading `migrate-prim-based.rst`. The code examples here are **not** primitive routines, since they don't use primitives. Instead, they are algorithms that use primitive routines. Which makes this how-to very confusing. 

I don't quite understand why there's a how-to on using primitive routines. If you're writing an algorithm that uses a fidelity routine, you shouldn't care _how_ the routine was written (as long as it has the expected interface). It's the person who has to write said routine that needs to care.

Reply:
Will discuss with Tushar and Dariusz